Dockerfile

```diff
@@ -1 +1,82 @@
 
+# Define the base image tag version at build time
+ARG UBUNTU_VERSION=20.04
+
+# Use Ubuntu as the base image with the specified version
+FROM ubuntu:${UBUNTU_VERSION}
+
+# Set the image maintainer/owner
+LABEL maintainer="DEVOPS EASY LEARNING"
+
+# Install required packages
+RUN apt-get update && apt-get install -y \
+    ansible \
+    curl \
+    git \
+    gnupg \
+    jq \
+    linux-headers\
+    openssh-client \
+    postgresql-client \
+    python3 \
+    kubectl \
+    kubens \
+    nodejs \
+    npm \
+    vim \
+    wget \
+    python3-pip \
+    net-tools \
+    iputils-ping \
+    terraform \
+    awscli \
+    default-jre \
+    default-jdk \
+    maven \
+    helm \
+    ufw \
+    go && \
+    rm -rf /var/lib/apt/lists/*
+
+# Set the default directory
+WORKDIR /BUILDER
+
+# Create environment variables (APP_NAME and ENV) defined at build time
+ARG APP_NAME
+ARG ENV
+ENV APP_NAME=$APP_NAME
+ENV ENV=$ENV
+ENV TEAM=Devops
+
+# Set the default user to root
+USER root
+
+# Create the REPOS directory under the root user's home directory
+RUN mkdir -p /root/REPOS
+
+# Open port range from 80 to 6000 (excluding 3030, 4878, 4596)
+EXPOSE 80-6000/tcp
+EXPOSE 3030/tcp
```

Comment:
EXCLUDE 3030,4878,4596

Reply:
I agree with s6nic that you need to exclude those ports as you expose them. Great job so far.